package cit.backen.library.management.system.loan.controller;


import cit.backen.library.management.system.api.response.ApiResponse;
import cit.backen.library.management.system.loan.dto.LoanRequest;
import cit.backen.library.management.system.loan.dto.LoanResponse;
import cit.backen.library.management.system.loan.service.LoanService;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/v1/loans")
public class LoanController {
    private final LoanService loanService;

    public LoanController(LoanService loanService) {
        this.loanService = loanService;
    }

    @PostMapping
    public ResponseEntity<ApiResponse<LoanResponse>> addLoan(@Valid @RequestBody LoanRequest request){
        ApiResponse<LoanResponse> apiResponse = loanService.addLoan(request);
        return ResponseEntity.status(HttpStatus.CREATED).body(apiResponse);
    }
}
